static/app/views/settings/organizationAuthTokens/newAuthToken.tsx

```diff
@@ -1,14 +1,169 @@
-import EmptyMessage from 'sentry/components/emptyMessage';
+import {useState} from 'react';
+import {browserHistory} from 'react-router';
+import styled from '@emotion/styled';
+
+import {addErrorMessage, addSuccessMessage} from 'sentry/actionCreators/indicator';
+import Alert from 'sentry/components/alert';
+import {Button} from 'sentry/components/button';
+import {Form, TextField} from 'sentry/components/forms';
+import FieldGroup from 'sentry/components/forms/fieldGroup';
 import ExternalLink from 'sentry/components/links/externalLink';
-import {Panel, PanelBody, PanelHeader} from 'sentry/components/panels';
+import {Panel, PanelBody, PanelHeader, PanelItem} from 'sentry/components/panels';
 import SentryDocumentTitle from 'sentry/components/sentryDocumentTitle';
+import TextCopyInput from 'sentry/components/textCopyInput';
 import {t, tct} from 'sentry/locale';
-import {Organization} from 'sentry/types';
+import {Organization, OrgAuthToken} from 'sentry/types';
+import getDynamicText from 'sentry/utils/getDynamicText';
+import {handleXhrErrorResponse} from 'sentry/utils/handleXhrErrorResponse';
+import {useMutation, useQueryClient} from 'sentry/utils/queryClient';
+import RequestError from 'sentry/utils/requestError/requestError';
+import useApi from 'sentry/utils/useApi';
+import {normalizeUrl} from 'sentry/utils/withDomainRequired';
 import withOrganization from 'sentry/utils/withOrganization';
 import SettingsPageHeader from 'sentry/views/settings/components/settingsPageHeader';
 import TextBlock from 'sentry/views/settings/components/text/textBlock';
+import {makeFetchOrgAuthTokensForOrgQueryKey} from 'sentry/views/settings/organizationAuthTokens';
+
+type CreateTokenQueryVariables = {
+  name: string;
+};
+
+type OrgAuthTokenWithToken = OrgAuthToken & {token: string};
+
+type CreateOrgAuthTokensResponse = OrgAuthTokenWithToken;
+
+function AuthTokenCreateForm({
+  organization,
+  onCreatedToken,
+}: {
+  onCreatedToken: (token: OrgAuthTokenWithToken) => void;
+  organization: Organization;
+}) {
+  const initialData = {
+    name: '',
+  };
+
+  const api = useApi();
+  const queryClient = useQueryClient();
+
+  const goBack = () => {
+    browserHistory.push(normalizeUrl(`/settings/${organization.slug}/auth-tokens/`));
+  };
```

Comment:
```suggestion
 const handleGoBack = useCallback(() => {
    browserHistory.push(normalizeUrl(`/settings/${organization.slug}/auth-tokens/`));
 }, [organization.slug])
```

otherwise this function will be created on every render